cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/VerificationStage.java

```diff
@@ -169,6 +170,19 @@ public void process(ApplicationDeployable input) throws Exception {
       }
     }
 
+    // TODO StreamSizeSchedules should be resilient to stream inexistence [CDAP-1446]
+    for (Map.Entry<String, ScheduleSpecification> entry : specification.getSchedules().entrySet()) {
+      Schedule schedule = entry.getValue().getSchedule();
+      if (schedule instanceof StreamSizeSchedule) {
+        StreamSizeSchedule streamSizeSchedule = (StreamSizeSchedule) schedule;
+        String streamName = streamSizeSchedule.getStreamName();
+        if (!specification.getStreams().containsKey(streamName)) {
```

Comment:
Is this accurate? I am not sure what is allowed... I see two valid cases:

1. this app creates the stream, then it is in the spec of this app
2. another app created the stream, and it already exists. In this case, it would not necessarily be in the app spec. You would have to check whether the stream already exists.